manager.py

```diff
@@ -281,8 +281,13 @@ def run(self):
 
                 self.handle_cpm(this_start, this_end)
                 if self.quit_after_interval:
-                    sys.exit(0)
+                    self.vprint(1, 'Reboot: taking down Manager')
+                    self.stop()
+                    self.takedown()
+                    os.system('sudo shutdown now -r')
                 this_start, this_end = self.get_interval(this_end)
+                if time.altzone < self.interval/60:
```

Comment:
So you are trying to use `time.altzone` to trigger a reboot at midnight. I still question whether this does what you think it does because `altzone` is a timezone that doesn't change.

But it's a moot point because I realized that the nightly reboot should still be handled by `crontab`. `crontab` will send a `SIGQUIT` signal to python. So `SIGQUIT` will trigger `self.quit_after_interval` which will run `self.takedown()` and run `git-pull-reboot.sh` which is all good.

So I think you can remove these two lines of code.